yt/geometry/grid_geometry_handler.py

```diff
@@ -475,7 +475,16 @@ def _mesh_sampling_particle_field(field, data):
 
             i, j, k = np.floor((pos - data.LeftEdge) / data.dds).astype("int64").T
 
-            return field_values[i, j, k]
+            # Make sure all particles are within the current grid, otherwise return nan
+            maxi, maxj, maxk = field_values.shape
+
+            mask = (i < maxi) & (j < maxj) & (k < maxk)
+            mask &= (i >= 0) & (j >= 0) & (k >= 0)
+
+            result = np.full_like(i, np.nan, dtype="float64")
+            result[mask] = field_values[i[mask], j[mask], k[mask]]
```

Comment:
Isn't this equivalent ? (if it's not, disregard)
```suggestion
            result[mask] = field_values[i, j, k][mask]
```

Reply:
It is not, as there may be some values of `i`, `j` or `k` that are out-of-bounds (so `field_values[i, j, k]` will fail).

Reply:
Ok, I wasn't sure. Thanks !